### DISCUSSION TITLE
Internal Server Error: 500

### DISCUSSION THREAD

**Discussion by tomku777** [microsoft-search/pnp-modern-search#1877] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/6361843/155975360-7c5fb633-876a-4541-8394-456aba0e8ff0.png)

Hi all, please, is there any place I can go and check what happens? Strange is that it sometimes work and sometimes not. And I don't know why.

**Comment by wobba** [comment]
No, this is an internal error coming from Microsoft. There are issues reported related to query rules and people search seeing similar errors, but they are API related, not web part related. Without you sharing anything about your web part setup in terms of which search API you are using or what the query is it's impossible to have an educated guess.

**Reply by tomku777** [reply]
Hi Mikael! Thank you for your reply! 

Basically, we are using query rules to filter search results just for one specific list and its items. Then we do visual template customization. Query template is: {queryTerms} Path:"https://contoso.sharepoint.com/lists/TempList" contenttypeid:0x01*

As described earlier sometimes it works and sometimes not. Sometimes it needs few refreshes and voila, it works.

**Reply by wobba** [reply]
Random bugs are not easy to solve :( Do you have a support case open, and can you try to replicate the setup using classic web parts?

**Reply by LetMeSP** [reply]
We are having the same issue. Seems very random.

**Comment by Namrata99sah** [comment]
[HTTP]:500 - [CorrelationId]:09ad28a0-b014-3000-ed9b-65ceaddae771 [Version]:16.0.0.22223 this is happening with me when filter webpart is getting added.

**Reply by wobba** [reply]
You need to look at the http request/response. If the request is correct, then it's an API issue. If you are not using query rules, then you should log a separate issue. But you still need to provide good debug information if we are to provide an answer. So far, these 500 errors have all been API related and should be logged with Microsoft.

**Comment by patrikhellgren** [comment, marked as answer]
We started seeing constant 500 errors in the results webpart about a week ago in a customers setup. It had been working for several months and no changes had been made.
After some testing a colleague of mine found that there was a property called DocumentSummary selected in the drop down for Selected Properties in the property pane of the webpart. After deselecting that it all started working again.
If I now check in the drop down I cannot find that property again and if I try a query with Search Query Tool and specify DocumentSummary as a selected property I get a 500 error there also. We get this in multiple tenants so there seems like there has been some changes in the api that doesn't like the DocumentSummary property even though it exists as a managed property in the search schema. We haven't opened a support case for this since we don't really need it at the moment so I just thought I would share it here if someone else ran into the same issue.

**Reply by tomku777** [reply]
Thank you!!! Will try it and let you know!

**Reply by wobba** [reply]
Interesting and `DocumentSummary` at least is not part of props we request by default for either v3 or v4. I know these have generated issued in the past: "ClassificationLastScan", "ClassificationConfidence", "ClassificationCount", "ClassificationContext"

**Reply by Namrata99sah** [reply]
In My Case below combination works -
1. Disabling Query Rules from WP config
2. Enable Query Rules but Need to remove Filter/Refiner WP

Not Sure if an issue with Filter Webpart

**Reply by tomku777** [reply]
Thank you guys! In our case DocumentSummary was not the issue. For now the solution was to clear all properties that are not used by our template and now it works as expected! Let's see after few days but again, thank you - your help was valuable for us.

**Comment by LetMeSP** [comment]
I created a ticket for Microsoft about this issue and after sending some traces I got this response from their Sr. Support Escalation Engineer: "Our product group is working for a fix on this intermittent failure and the fix will be rolled out in couple of weeks."  I will keep you posted.

**Reply by LetMeSP** [reply]
I received this today from Microsoft:
"The issue is resolved by our product group with a fix last week, but you will see random 500 error which will be very less in frequency. Our product group said that these 500 errors are transient and will occur with low frequency, a new request should go through. I am not sure if this random 500 errors will go away in the future when new updates are made to search service.  But that is a possibility."

**Reply by wobba** [reply]
Let's hope they will be very infrequent.
